src/data/tts.py

```diff
@@ -161,3 +84,41 @@ def generate(self, text: str, speaker: str = "alloy") -> np.array:
         audio = self.convert24to16khz(audio).numpy()
         os.remove(self.cached_file)
         return audio.flatten()
+
+
+class SileroTTSProvider(BaseTTSProvider):
+    lang_to_model_id = dict(
+        ru="v4_ru",
+        ua="v4_ua",
+        en="v3_en",
+    )
+
+    def __init__(
+            self,
+            language: str,
+            silero_rate: int = 24000,
+            gen_rate: int = 16000,
+    ):
+        """
+        :param language: can be "ru" (6 speakers) or "ua" (2 speakers) "en" (a lot of speakers)
+        :param silero_rate: the rate for silero model audio output, can be [48000, 24000, 8000]
+        :param gen_rate: the generator will resample the audio to this rate
+        """
+
+        self.model, _ = torch.hub.load(repo_or_dir='snakers4/silero-models',
+                                       model='silero_tts',
+                                       language=language,
+                                       speaker=self.lang_to_model_id[language])
+        self.silero_rate = silero_rate
+        self.gen_rate = gen_rate
+        self.speaker_mapper = SpeakerMapper(speakers_bank=self.model.speakers)
+
+    def generate(self, text: str, speaker: str) -> np.array:
+        if not text:
+            return
+        speaker = self.speaker_mapper.map(speaker)
+        audio = self.model.apply_tts(text=text,
+                                speaker=speaker,
+                                sample_rate=self.silero_rate)
+        audio = librosa.resample(audio.numpy(), orig_sr=self.silero_rate, target_sr=self.gen_rate)
```

Comment:
maybe torch.Resample

Reply:
Yeah, that sounds cleaner. I will change, but I will not be surprised if it uses librosa internally

Reply:
no torch has own implementation